packages/helpers/src/validate.ts

```diff
@@ -0,0 +1,116 @@
+import { BytesLike, number } from "@ckb-lumos/codec";
+import { createTransactionFromSkeleton, TransactionSkeletonType } from ".";
+import { bytify, hexify } from "@ckb-lumos/codec/lib/bytes";
+import blake2b from "blake2b";
+import pick from "lodash.pick";
+import { blockchain } from "@ckb-lumos/base";
+
+const { RawTransaction } = blockchain;
+
+type HashAlgorithm = "ckb-blake2b-256" | "ckb-blake2b-160";
+
+interface Hasher {
+  update(data: BytesLike): Hasher;
+  digest(): string;
+}
+
+class Blake2bHasher implements Hasher {
+  constructor(
+    outlength: number,
+    key: Uint8Array | undefined,
+    salt: Uint8Array | undefined,
+    personal: Uint8Array | undefined
+  ) {
+    this.hasher = blake2b(outlength, key, salt, personal);
+  }
+  private hasher: blake2b.Blake2b;
+
+  update(data: BytesLike) {
+    this.hasher.update(bytify(data));
+    return this;
+  }
+
+  digest() {
+    return hexify(this.hasher.digest());
+  }
+}
+
+function createHasher(algorithm: HashAlgorithm): Hasher {
+  switch (algorithm) {
+    case "ckb-blake2b-256":
+      return new Blake2bHasher(
+        32,
+        undefined,
+        undefined,
+        Buffer.from("ckb-default-hash")
+      );
+
+    case "ckb-blake2b-160":
+      return new Blake2bHasher(
+        20,
+        undefined,
+        undefined,
+        Buffer.from("ckb-default-hash")
+      );
+    default:
+      throw new Error(`Unsupported hash algorithm: ${algorithm}`);
+  }
+}
+
+function getWitnessLength(witnesses: Uint8Array) {
+  return number.Uint64.pack(witnesses.length).buffer;
+}
+
+/**
+ * Validate an transaction.
+ * @param messagesForSigning the message for signing. Each item in it means hash(rawTransaction | extraData).
+ * @param rawTransaction raw transaction object
+ * @param extraData the extra data for validate
```

Comment:
content to be hashed other than `rawTransaction`, is generally processed witness